components/entrytypes.py

```diff
@@ -0,0 +1,511 @@
+        return self._repository.name
+
+    @property
+    def sha(self) -> str:
+        return self._commit.sha
+
+    @property
+    def short_sha(self) -> str:
+        return self.sha[:7]
+
+    @property
+    def url(self) -> str:
+        return self._commit.html_url
+
+    @property
+    def title(self) -> str:
+        return self._commit.commit["message"]
+
+    @property
+    def author(self) -> str:
+        return self._commit.author["login"]
+
+    @property
+    def short_name(self) -> str:
+        return (
+            f'{"" if self.owner == DEFAULT_REPO_OWNER else self.owner + "/"}'
+            f'{"" if self.repo == DEFAULT_REPO_NAME else self.repo}'
+            f"@{self.short_sha}"
+        )
+
+    @property
+    def display_name(self) -> str:
+        return f"Commit {self.short_name}: {self.title} by {self.author}"
+
+    @property
+    def description(self) -> str:
+        return "Search on GitHub"
+
+    def html_markup(self, _: str = None) -> str:
+        return f'<a href="{self.display_name}">{self.url}</a>'
+
+    def html_insertion_markup(self, _: str = None) -> str:
+        return f'<a href="{self.short_name}">{self.url}</a>'
+
+    def compare_to_query(self, search_query: str) -> float:
+        search_query = search_query.lstrip("@ ")
+        if self.sha.startswith(search_query):
+            return 100
+        return 0
+
+
+class Issue(BaseEntry):
+    """An issue/PR on Github
+
+    Args:
+        issue: The github3 issue object
+        repository: The github3 repository object
+    """
+
+    def __init__(self, issue: GHIssue, repository: GHRepo) -> None:
+        self._issue = issue
+        self._repository = repository
+
+    @property
+    def type(self) -> str:
+        return "Issue" if not self._issue.pull_request_urls else "PR"
+
+    @property
+    def owner(self) -> str:
+        return self._repository.owner.login
+
+    @property
+    def repo(self) -> str:
+        return self._repository.name
+
+    @property
+    def number(self) -> int:
+        return self._issue.number
+
+    @property
+    def url(self) -> str:
+        return self._issue.html_url
+
+    @property
+    def title(self) -> str:
+        return self._issue.title
+
+    @property
+    def author(self) -> str:
+        return self._issue.user.login
+
+    @property
+    def short_name(self) -> str:
+        return (
+            f'{"" if self.owner == DEFAULT_REPO_OWNER else self.owner + "/"}'
+            f'{"" if self.repo == DEFAULT_REPO_NAME else self.repo}'
+            f"#{self.number}"
+        )
+
+    @property
+    def display_name(self) -> str:
+        return f"{self.type} {self.short_name}: {self.title} by {self.author}"
+
+    @property
+    def description(self) -> str:
+        return "Search on GitHub"
+
+    @property
+    def short_description(self) -> str:
+        string = f"{self.type} {self.short_name}: {self.title}"
+        return (string[:25] + "…") if len(string) > 25 else string
```

Comment:
unfortunately not - that would give circular imports since we use `TAG_HINTS` in `utils.buld_command_list`. I think I even tried with importing `truncate_str` only locally …

Reply:
mh, apparently I did not try …